protocol/0062-SPAM-spam_protection.md

```diff
@@ -83,6 +83,42 @@ Further, each party is allowed to submit up to `n` transactions per epoch where
+
+Before any order or liquidity commitment is accepted for a perpetual futures or expiring futures check that the party has `margin + order margin + general > 0` with `margin` being the balance in the margin account for the relevant, `order margin` is the balance for the order margin if party is in isolated margin mode for the for the relevant market and `general` the balance in the general account for the relevant asset. Orders from parties that don't meet this criteria are rejected. This is to be done after the PoW check.
+
+Further, given the network parameter: `spam.order.minimalMarginQuantumMultiple` (between 0 and infinite)
+
+If the maintenance margin for a given transaction is smaller than the parameter `spam.order.minimalMarginQuantumMultiple`, then the transaction is pre-block rejected.
+I.e. if `(rf + linear slippage param) x size x price <  spam.order.minimalMarginQuantumMultiple x asset quantum amount` then the order is rejected. Here `rf` is the risk factor (and will be different for long and for short) `linear slippage param` is a market parameter and `size` and `price` are assumed to be correctly scaled by, PDPs and MDPs respectively. 
+
+For pegged orders `mark price +/- offset` should be used in place of price. The `mark price` value can be cached from end of last block (if that helps for performance, we need this to be cheap). In opening auctions pegged orders should be rejected.
+
+For stop / position linked orders: we don't necessarily have either price or size. However there is a limit on the number per key so we let these through.  
+
+Finally, if the market does not exist and thus the maintenance margin is not defined, the transaction is rejected.
+
+#### Spot markets
+
+Before any order or liquidity commitment is accepted for a spot market, check that the party has balances where `holding + general > 0` for the asset that the order is potentially disposing of / committing in terms of liquidity. Orders from parties that don't meet this criteria are rejected. This is to be done after the PoW check.
+
+Further, given the network parameter: `spam.order.minimalHoldingQuantumMultiple` (between 0 and infinite)
+
+If the holding requirement for a given transaction is smaller than the parameter `spam.order.minimalHoldingQuantumMultiple`, then the transaction is pre-block rejected.
+I.e. if `size x price <  spam.order.minimalHoldingQuantumMultiple x quote asset quantum amount` then the order is rejected. Here `size` and `price` are assumed to be correctly scaled by, PDPs and MDPs respectively. 
+
+For pegged orders `last traded price +/- offset` should be used in place of price. The `last traded price` value can be cached from end of last block (if that helps for performance). In opening auctions pegged orders should be rejected.
+
+#### All markets
+
+The following points apply generally to both derivative and spot markets:
+
+- Non-persistent orders are not blocked by the mechanisms described above as long as the market exists.
```

Comment:
What exactly do we mean by this? Is it that market orders are not subject to the minimum size checks? @cdummett

Reply:
Resolving as PR updated since them and comment outdated.